src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -2552,9 +2552,6 @@ static void M1(B b)
             // https://github.com/dotnet/roslyn/issues/44691
             var comp = CreateCompilation(new[] { src, AllowNullAttributeDefinition });
             comp.VerifyDiagnostics(
-                // (12,26): warning CS8625: Cannot convert null literal to non-nullable reference type.
-                //         b = b with { X = null }; // ok
-                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(12, 26),
```

Comment:
>Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(12, 26), [](start = 16, length = 73)

This change is unexpected. I assume that is because we bind parameter modifiers on properties right now. I do not believe this is something we even want in the future.